### PR TITLE
Question editing   server

### PR DESCRIPTION
**Pull request by ReemCohen**

### CODE REVIEW

File: server/src/DAL/questionsRepository.ts [hunk 1]
```diff
@@ -49,14 +49,13 @@ class DBQuestionsRepository {
...
-		console.log(this.data);
-
-		//need to complete
+		let index = this.data?.findIndex(q => q.id === id);
+		if (index === -1 || index === undefined) {
+			throw new ItemNotInDbError(id, "Question");
+		}
+		if (this.data) {
+			this.data[index] = { ...this.data[index], ...question };
+		}
```
Comment (ghost) [inline server/src/DAL/questionsRepository.ts]: This condition covers all the problems, in one condition.
In addition, this saves the data and returns the saved element.
Also, we make sure we don't use a wrong Id in no case.

```suggestion
		if (!this.data || !index || index < 0) {
			throw new ItemNotInDbError(id, "Question");
		}
		this.data[index] = { ...this.data[index], ...question, id };
		this.writeToFile(this.data);
		return this.data[index];
```

File: server/src/DAL/questionsRepository.ts [hunk 2]
```diff
@@ -49,15 +49,25 @@ class DBQuestionsRepository {
...
 		}
 		this.data[index] = { ...this.data[index], ...question, id };
-		this.writeToFile(this.data);
+		await this.writeToFile(this.data);
 		return this.data[index];
 	}
 
+	async deleteQuestion(id: models.classes.guid) {
+		let index = (this.data || []).findIndex(q => q.id === id);
+		if (!this.data || index < 0) {
```
Comment (ReemCohen) [inline server/src/DAL/questionsRepository.ts]: Shouldn't we need to define _findIdex_ method as well as  validation check for _this.data_ and _index_ in 
seperate method?

as you said before - if it repeat more than once the code isn't in his best shape

Comment (ghost) [reply server/src/DAL/questionsRepository.ts]: That is a really good idae 🧙🏾‍♂️

Comment (ghost) [reply server/src/DAL/questionsRepository.ts]: @ReemCohen Please check out [the fix](https://github.com/ChemiAtlow/YahalomTests/pull/29/commits/04cf1bf1541861ad0e4a1fb91490c2264d7d8ed2)

**Review by ghost — APPROVED** [review]
LGTM

**Pull request merged**